Simplenote/Classes/SPMarkdownParser.m

```diff
@@ -29,22 +29,40 @@ + (NSString *)renderHTMLFromMarkdownString:(NSString *)markdown
     hoedown_html_renderer_free(renderer);
     
     NSString *htmlString = [[NSString alloc] initWithData:htmlData encoding:NSUTF8StringEncoding];
+    
+    NSRegularExpression *regex = [NSRegularExpression regularExpressionWithPattern:
+                                  @"\n\\s+- \\[(x|X|o|o| )\\]\\s+([^\n]+)" options:0 error:nil];
```

Comment:
Definitely. Unfortunately it doesn't even do what I need it to do, which is link the rendered text to the plaintext in the note. In JavaScript I was able to do this because the RegExp replace takes a callback where I could track the match index in the full string. Here however I think I need to actually break this apart into a loop and build the string incrementally.